### ISSUE TITLE
[Suggestion] Real-Time Page Rendering

### ISSUE THREAD

**Issue by zurgeg** [tenox7/wrp#78]
In my opinion, WRP should at least have an option to render the page every time it is updated, this would make the page much more responsive. You could also make it run at 1 frame per x seconds by putting a JS refresh (for some newer devices) every x seconds. Thanks in advance.

**Comment by tenox7** [comment]
please elaborate on when is it that a page is updated?

but I do see a need to add a button to re-capture page with a button
for auto refresh maybe http-equiv refresh, I don't like javascript, but maybe

**Comment by zurgeg** [comment]
For instance, on youtube, whenever a new frame is pushed @tenox7.

**Comment by tenox7** [comment]
that would probably require some streaming protocol, you don't want to be generating a new png/gif image every frame

hmm maybe animated gif?

**Comment by zurgeg** [comment]
@tenox7 That might work.

**Comment by zurgeg** [comment]
I think some older systems (like Windows 95) might not like that.

**Comment by wojtekpolska** [comment]
@tenox7 I'd really enjoy if you added 1 (or 2) buttons: 
- (the more important): as you suggested, a re-capture image - some websites (like discord) are loading a bit longer than it takes to take a screenshot, so in the browser window i just see an image of loading screen. Currently the work-around is clicking something like the Backspace button, but if the website uses all buttons for something, then there is a problem. 
- (less important): normal refresh - it will make a page reload on host, and then send a new image. Currently the work-around is just clicking GO again keeping the same link

**Comment by zurgeg** [comment]
@wojtekpolska I think that there should be an option for streaming, for some of the newer browsers that support it, also, I think an animated GIF would be handy for some browsers "in the middle" that support it.

**Comment by tenox7** [comment]
Agreed. I will definitely add "re-capture image" option/button without page refresh. As for animated gif I'm going to do some experiments and see whats doable.

**Comment by wojtekpolska** [comment]
@tenox7 Thanks! I like this tool already, im using it out of curiosity, but im pretty sure that some legacy systems might heavily benefit from it :)

PS. The Readme file is missing an explanation to "S" function and im not really sure what it does either, i assume its size, but why is it needed when we already choose the page size in px? 

If you have time please add information about "S" to the Readme

**Comment by wojtekpolska** [comment]
@tenox7 Thanks! I like this tool already, im using it out of curiosity, but im pretty sure that some legacy systems might heavily benefit from it :)

PS. The Readme file is missing an explanation to "S" function and im not really sure what it does either, i assume its size, but why is it needed when we already choose the page size in px? it probably has some use that im not aware of

If you have time please add information about "S" to the Readme

**Comment by tenox7** [comment]
S is for scale, or zoom. Thanks for pointing it out!! I'll fix it in the readme.

**Comment by wojtekpolska** [comment]
Im happy to be helpful :D

**Comment by tenox7** [comment]
I have added "recapture image" option. This is done with **St** button which basically sends "Stop" to chrome and then recaptures page. This even works in Youtube, you can play a video and click on St and it will refresh it on the client without reloading whole page.

**Comment by tenox7** [comment]
@wojtekpolska I have also renamed S (scale) to Z (zoom). I checked with Chrome and Firefox and they both have "Zoom" instead of scale so this should be more clear now.

**Comment by wojtekpolska** [comment]
@tenox7 how to download the latest version? The latest release is from 2019, i see the commits but im not sure how to download it

**Comment by zurgeg** [comment]
@tenox7 I came up with an idea. Would using a small JS program to auto-query for a new frame work? I'm pretty sure that's what browservice does.

**Comment by tenox7** [comment]
I have a method for recapturing screenshot without page refresh. There is a button for this in the default html. It's not released but it's in head main (I will be releasing shortly). You could trigger this with JavaScript easily. Saying that I dont want to add JS to the default html but now you can customize it yourself :)

**Comment by tenox7** [comment]
this is now released as 4.5.2
https://github.com/tenox7/wrp/releases/tag/4.5.2
